TrackMania 2D/src/ia/ga/impl/string/StringCalculator.java

```diff
@@ -0,0 +1,38 @@
+package ia.ga.impl.string;
+
+import java.util.List;
+
+import ia.ga.core.FitnessCalc;
+import ia.ga.core.Individual;
+
+public class StringCalculator implements FitnessCalc<Character> {
+
+	String solution = "jesuissuper";
+
+	@Override
+	public Integer getFitness(Individual<Character> individual) {
+		return solution.length() - hammingDist(solution.toCharArray(), convertToCharArray(individual.getChromosome()));
+	}
+
+	@Override
+	public Integer getMaxFitness() {
+		return solution.length();
+	}
+
+	private static int hammingDist(char str1[], char str2[]) {
+		int count = 0;
+		for (int i = 0; i < str1.length; i++) {
+			if (str1[i] != str2[i])
+				count++;
+		}
```

Comment:
c'est pas la bonne implémentation de la distance de hamming (en plus levenstein est plus pertinant que hamming pour les comparaisons de chaine de character)

Reply:
C'était vraiment pour faire un exemple qui marche :) cependant je ne vois pas pourquoi ce n'est pas la bonne implémentation de la distance de hamming (d'ailleurs je pense que c'est l'implémentation la plus simple pour deux string de même longueur)

Reply:
hamming compare des ensemble d'élément sans tenir compte de l'ordre, avec hamming on peut avoir une distance de 0 entre "hamming" et "mimghan"

Reply:
en.m.wikipedia.org/wiki/Hamming_distance - >2173896 and 2233796 is 3.

Reply:
je suis allé relire, et je m'incline -_-